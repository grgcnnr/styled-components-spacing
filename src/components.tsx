import styled from 'styled-components';
import {ThemedValueOrValueMap} from './types';
import {
  ThemedMarginProps,
  ThemedPaddingProps,
  m,
  mx,
  my,
  mt,
  mr,
  mb,
  ml,
  p,
  px,
  py,
  pt,
  pr,
  pb,
  pl,
  marginProps,
  paddingProps,
} from './themed';

interface InlineProps {
  inline?: boolean;
}

const inlineMixin = ({inline}: InlineProps): string | undefined =>
  (inline && 'display: inline-block;') || undefined;

export type DeprecatedSpacingProps = InlineProps & {
  all?: ThemedValueOrValueMap;
  horizontal?: ThemedValueOrValueMap;
  vertical?: ThemedValueOrValueMap;
  top?: ThemedValueOrValueMap;
  right?: ThemedValueOrValueMap;
  bottom?: ThemedValueOrValueMap;
  left?: ThemedValueOrValueMap;
};

/** @deprecated */
export const Margin = styled.div<DeprecatedSpacingProps>`
  ${inlineMixin}
  ${({all}) => all && m(all)}
  ${({horizontal}) => horizontal && mx(horizontal)}
  ${({vertical}) => vertical && my(vertical)}
  ${({top}) => top && mt(top)}
  ${({right}) => right && mr(right)}
  ${({bottom}) => bottom && mb(bottom)}
  ${({left}) => left && ml(left)}
`;

Margin.displayName = 'Margin';

/** @deprecated */
export const Padding = styled.div<DeprecatedSpacingProps>`
  ${inlineMixin}
  ${({all}) => all && p(all)}
  ${({horizontal}) => horizontal && px(horizontal)}
  ${({vertical}) => vertical && py(vertical)}
  ${({top}) => top && pt(top)}
  ${({right}) => right && pr(right)}
  ${({bottom}) => bottom && pb(bottom)}
  ${({left}) => left && pl(left)}
`;

Padding.displayName = 'Padding';

export type SpacingProps = InlineProps & ThemedMarginProps & ThemedPaddingProps;

export const Spacing = styled.div<SpacingProps>`
  ${inlineMixin}
  ${marginProps}
  ${paddingProps}
`;

Spacing.displayName = 'Spacing';
