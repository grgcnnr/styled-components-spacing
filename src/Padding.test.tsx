import 'jest-styled-components';
import React from 'react';
import { render } from '@testing-library/react';
import { Padding } from './components';

describe('<Padding/>', () => {
  it('it should have no styles when no props are passed', () => {
    const { container } = render(<Padding />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('inline', () => {
    it('it should be set when the prop is true', () => {
      const { container } = render(<Padding inline={true} />);
      expect(container.firstChild).toMatchSnapshot();
      expect(container.firstChild).toHaveStyleRule('display', 'inline-block');
    });

    it('it should not be set when the prop is false', () => {
      const { container } = render(<Padding inline={false} />);
      expect(container.firstChild).not.toHaveStyleRule('display', 'inline-block');
    });
  });

  describe('all', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding all={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding all={{ mobile: 0, tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const { container } = render(<Padding all={{ tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
      expect(container.firstChild).toHaveStyleRule('padding', '0.5rem', {
        media: 'screen and (min-width:46.0625em)'
      });
      expect(container.firstChild).toHaveStyleRule('padding', '2rem', {
        media: 'screen and (min-width:74.6875em)'
      });
    });
  });

  describe('horizontal', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding horizontal={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding horizontal={{ mobile: 0, tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const { container } = render(<Padding horizontal={{ tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('vertical', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding vertical={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding vertical={{ mobile: 0, tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const { container } = render(<Padding vertical={{ tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('top', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding top={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding top={{ mobile: 0, tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const { container } = render(<Padding top={{ tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('right', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding right={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding right={{ mobile: 0, tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const { container } = render(<Padding right={{ tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('bottom', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding bottom={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding bottom={{ mobile: 0, tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const { container } = render(<Padding bottom={{ tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('left', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding left={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const { container } = render(<Padding left={{ mobile: 0, tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const { container } = render(<Padding left={{ tablet: 2, desktop: 4 }} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
