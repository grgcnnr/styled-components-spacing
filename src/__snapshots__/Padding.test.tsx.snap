// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Padding/> all it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> all it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> all it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Padding/> bottom it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-bottom: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> bottom it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> bottom it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-bottom: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Padding/> horizontal it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-left: 0;
  padding-right: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> horizontal it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> horizontal it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Padding/> inline it should be set when the prop is true 1`] = `
.c0 {
  display: inline-block;
}

<div
  class="c0"
/>
`;

exports[`<Padding/> it should have no styles when no props are passed 1`] = `
<div
  class=""
/>
`;

exports[`<Padding/> left it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-left: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-left: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-left: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> left it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-left: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-left: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> left it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-left: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Padding/> right it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-right: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> right it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> right it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-right: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Padding/> top it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-top: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-top: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-top: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> top it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-top: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-top: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> top it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-top: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Padding/> vertical it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-top: 0;
  padding-bottom: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-top: 2rem;
    padding-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> vertical it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-top: 2rem;
    padding-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Padding/> vertical it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

<div
  class="c0"
/>
`;
